provisioner/registryv1/README.md

```diff
@@ -0,0 +1,123 @@
+
+The registryv1 provisioner is able to unpack a given `registry+v1` bundle onto a cluster and then instantiate it, making the
+content of the bundle available in the cluster. It does so by reconciling the `Bundle` and `BundleInstance` types that 
+have the spec.provisionerClassName field set to `core.rukpak.io/registry+v1`. This field must be set to the correct provisioner name 
+in order for the registryv1 provisioner to see and interact with the bundle. 
+
+Since this provisioner is responsible for creating OLM bundles on-cluster, it has a soft dependency on 
+OLM and the existing set of APIs, specifically ClusterServiceVersion and OperatorGroup. Those APIs must exist on the cluster
+in order to instantiate a `registry+v1` bundle successfully. The provisioner creates the CSV provided in the bundle, but delegates
+additional handling of the CSV to OLM itself, therefore OLM should be installed on the cluster alongside this provisioner. 
+
+## Use cases
+
+### Install and apply a specific version of a bundle 
+
+The registryv1 provisioner can install and make available a specific `registry+v1` bundle available in the cluster. This is in contrast to 
+the existing OLM workflow which requires a Subscription and manual approval for the relevant InstallPlan to pin to a specific version.  
+
+Simply create a `Bundle` resource pointing to a specific version of your bundle, and a `BundleInstance` which references that bundle.
+The registryv1 provisioner will unpack the provided Bundle onto the cluster, and eventually make the content available on the cluster. 
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle-image@sha256:xyz123
+  provisionerClassName: core.rukpak.io/registry+v1
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/registry+v1
+  bundleName: my-bundle
+``` 
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak APIs, `Bundle` and `BundleInstance`. 
+A user can specify a particular `Bundle` to be available in the cluster for inspection before any application of the resources. 
+Given a `Bundle` resource named `my-bundle`, the registryv1 provisioner will pull down and unpack the bundle to a variety of storage backends, such as ConfigMaps or PersistentVolumes.
+
+Currently the ConfigMap backend is supported by the provisioner, and the content can be inspected by viewing the metadata ConfigMap, which holds
+information on all ConfigMaps related to this bundle. 
+
+`kubectl get cm bundle-metadata-my-bundle -o yaml` provides access to this metadata ConfigMap, which then links to all other ConfigMap objects
+where the unpacked data is stored. 
```

Comment:
There is a [label](https://github.com/operator-framework/rukpak/blob/7cc5fa45e61c2ef324bdbb5a8cf02bdaa42fc527/internal/storage/storage.go#L142) on the configmaps that contains the bundle's `.metadata.name`. That could be used in a label selector to find all of the configmaps.

One unrelated thing that comes to mind though: we might need to make the bundle configmap label key different than the[ label key used in the bundle instance reconciler](https://github.com/operator-framework/rukpak/blob/7cc5fa45e61c2ef324bdbb5a8cf02bdaa42fc527/provisioner/registryv1/controllers/bundleinstance_controller.go#L391).

Reply:
That label is helpful, I will add that information to the doc.